src/prefect/agent/kubernetes/agent.py

```diff
@@ -766,7 +769,13 @@ def generate_deployment_yaml(
         """
 
         # Use defaults if not provided
-        token = token or ""
+        if token:
```

Comment:
These changes seem unnecessary and overly risky -- these are all internal methods that still work correctly regardless of API Key or Token distinction

Reply:
> The docs changes are great; I'm less confident in all the code changes as many of the changes are for private methods that we don't need to take the risk with right now.
> 
> I think we should break this up into stages:
> 
> * docs
> * public APIs
> * internal APIs

Cool--I can do that.  Will push a PR for just docs.